enhancements/20240709-longhorn-v2-data-engine-v1.4.0.md

```diff
@@ -0,0 +1,174 @@
+# Support the Longhorn V2 Data Engine (v1.4.0, Experimental)
+
+## Summary
+
+The [Longhorn V2 Data Engine] has been available for preview use since Longhorn v1.5.0 and provides significantly better performance than the V1 Data Engine. Conseqently there is interest in being able to enable the V2 Data Engine in Harvester v1.4.0, which will be based on Longhorn v1.7.0. The V2 Data Engine will still be in preview or experimental status at that time (see the [Longhorn Roadmap]) so use of it in Harvester will also initially be considered experimental, and it will not be enabled by default.
+
+[Longhorn V2 Data Engine]: https://longhorn.io/docs/1.6.2/v2-data-engine/
+[Longhorn Roadmap]: https://github.com/longhorn/longhorn/wiki/Roadmap
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/5274
+
+## Motivation
+
+To give users who wish to enable the V2 Data Engine the ability to do so easily, via the Harvester GUI and/or installer, rather than having to make a series of potentially complicated and error-prone manual changes to the cluster.
+
+### Goals
+
+- Allow the V2 Data Engine to be enabled/disabled globally for the whole cluster
+- Allow the V2 Data Engine to be enabled/disabled on specific nodes
```

Comment:
From the user's viewpoint, he is going to enable the feature on all nodes with the `longhorn-v2-data-engine-enabled` setting ([PR](https://github.com/harvester/harvester/pull/6405)). Technically it's still possible to write NodeConfig for a specific node.